### PR TITLE
stop connection when disposing service hub context

### PR DESCRIPTION
**Pull request by realwanpengli**

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Management/ServiceHubContext.cs [hunk 1]
```diff
@@ -28,14 +28,28 @@ public ServiceHubContext(IHubContext<Hub> hubContext, IHubLifetimeManagerForUser
 
         public void Dispose()
         {
+            if (GetServiceConnectionStatus() != ServiceConnectionStatus.Disconnected)
+            {
+                StopConnectionAsync().Wait();
```
Comment (davidfowl) [inline src/Microsoft.Azure.SignalR.Management/ServiceHubContext.cs]: No please. Never use .Wait or .Result.

Comment (realwanpengli) [reply src/Microsoft.Azure.SignalR.Management/ServiceHubContext.cs]: OK. How about `_ = StopConnectionAsync()`

Comment (davidfowl) [reply src/Microsoft.Azure.SignalR.Management/ServiceHubContext.cs]: Nope, if you need to do asynchronous work, then await the result. Remove the IDisposable implementation and add a DisposeAsync that callers can use.

Comment (realwanpengli) [reply src/Microsoft.Azure.SignalR.Management/ServiceHubContext.cs]: OK. I will remove the interface `IDisposable` and its implementation and keep `DisposeAsync`.

File: src/Microsoft.Azure.SignalR.Management/ServiceHubContext.cs [hunk 2]
```diff
@@ -26,15 +25,25 @@ public ServiceHubContext(IHubContext<Hub> hubContext, IHubLifetimeManagerForUser
...
             _serviceProvider?.Dispose();
         }
 
-        public Task DisposeAsync()
+        private Task StopConnectionAsync()
         {
-            // todo: stop service connection
-            Dispose();
+            var serviceContainer = _serviceProvider.GetService<IServiceConnectionContainer>();
+            if (serviceContainer == null)
```
Comment (realwanpengli) [inline src/Microsoft.Azure.SignalR.Management/ServiceHubContext.cs]: In Transient mode, no service of `IServiceConnectionContainer`

**Pull request merged**